demos/generomak_solps_from_raw_files.py

```diff
@@ -0,0 +1,234 @@
+
+plt.figure()
+plt.imshow(c1_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CII density [m-3]')
+
+plt.figure()
+plt.imshow(c2_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CIII density [m-3]')
+
+plt.figure()
+plt.imshow(c3_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CIV density [m-3]')
+
+plt.figure()
+plt.imshow(c4_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CV density [m-3]')
+
+plt.figure()
+plt.imshow(c5_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CVI density [m-3]')
+
+plt.figure()
+plt.imshow(c6_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CVII density [m-3]')
+
+plt.figure()
+plt.imshow(h1_speed.T, extent=[xl, xu, zl, zu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('H+ speed [m/s]')
+
+plt.figure()
+plt.imshow(inside_samples.T, extent=[xl, xu, zl, zu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('Inside/Outside test')
+
+input('Press Return to continue...')
+plt.close('all')
+
+###############################################################################
+# Image the plasma with a camera.
+###############################################################################
+print('Imaging plasma...')
+world = World()
+plasma.parent = world
+
+# Cherab does not yet have a wall geometry specified for Generomak (see
+# https://github.com/cherab/core/issues/212), so we'll just make our own.
+# This will consist of a cylindrical center column and nothing else, which
+# will allow us to position an overview camera to image the whole plasma.
+centre_column_radius = 0.9 * me['minr']
+centre_column_height = me['maxz'] - me['minz']
+Cylinder(radius=centre_column_radius, height=centre_column_height,
+         transform=translate(0, 0, me['minz']),
+         material=AbsorbingSurface(), name='Centre column', parent=world)
+
+# A wide-angle pinhole camera looking horizontally towards the centre of the device.
+camera = PinholeCamera((128, 128))
+camera.parent = world
+camera.pixel_samples = 10
+camera.transform = (translate(2 * me['maxr'], 0, 0)
+                    * rotate_basis(Vector3D(-1, 0, 0), Vector3D(0, 0, 1)))
+camera.fov = 60
+
+# For each species in the plasma, model the total emission from that species.
+plasma.atomic_data = OpenADAS(permit_extrapolation=True)
+for species in plasma.composition:
+    if species.charge < species.element.atomic_number:
+        plasma.models.add(TotalRadiatedPower(species.element, species.charge))
```

Comment:
When and if raysect/source#343 is merged, I'll make a separate demo to show how the rendering time of this can be decreased by ~50 times.
```python
atomic_data = OpenADAS(permit_extrapolation=True)
emission = np.zeros((nx, 1, ny, camera.spectral_bins))
direction = Vector3D(0, 0, 1)
for species in plasma.composition:
    if species.charge < species.element.atomic_number:
        model = TotalRadiatedPower(species.element, species.charge, plasma=plasma, atomic_data=atomic_data)
        for i, x in enumerate(xsamp):
            for j, z in enumerate(zsamp):
                point = Point3D(x, 0, z)
                spectrum = Spectrum(camera.min_wavelength, camera.max_wavelength, camera.spectral_bins)
                emission[i, 0, j] += model.emission(point, direction, spectrum).samples

rgc = RegularGridCylinder(emission, spectrum.wavelengths, xu, zu - zl, radius_inner=xl, continuous=True,
                          parent=world, transform=translate(0, 0, zl))
```
In case you want to test this now, do not connect the plasma object to the world and increase the camera resolution to at least 512 x 512 to see a speedup.

Reply:
I think we have to decide where to pile the demo scripts. If we want to use the generomak in Cherab core as example data (which I thought was the purpose) then observation examples should be moved to the core package demos. The camera observation @jacklovell made could be kept in the solps package as an example for the sake of completeness.

Reply:
This would be better as one of a series of "performance optimisation" demos. There's a lot of added complexity in the fast emitter approach, which is a price worth paying if your render takes hours or days but is probably going to be overwhelming to a new Cherab user who doesn't mind waiting 1 minute for a render. I think we should keep the SOLPS demos as simple as possible and illustrate how the package fits into the standard Cherab workflow.

Reply:
Sure, it is definitely not for this demo.